### PR TITLE
Main request

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Copy of https://github.com/Sylius/SyliusResourceBundle/pull/332 to 1.8 branch

### CODE REVIEW

**Comment by Roshyo** [comment]
Isn't it a BC break ? I mean, some classes are not final, so introducing a return type should be BC break. Or am I wrong ?

**Comment by vvasiloi** [comment]
@Roshyo no, since PHP 7.4 there's full covariance support: https://www.php.net/manual/en/language.oop5.variance.php#language.oop5.variance.covariance

File: src/Bundle/Doctrine/ODM/MongoDB/DocumentRepository.php [hunk 1]
```diff
@@ -106,52 +96,43 @@ public function createPaginator(array $criteria = [], array $sorting = [])
...
+    protected function getQueryBuilder(): QueryBuilder
     {
         return $this->createQueryBuilder();
     }
 
-    /**
-     * @return QueryBuilder
-     */
-    protected function getCollectionQueryBuilder()
+    protected function getCollectionQueryBuilder(): QueryBuilder\
```
Comment (Roshyo) [inline src/Bundle/Doctrine/ODM/MongoDB/DocumentRepository.php:123]: Is that trailing `\` desired ?

File: src/Bundle/Doctrine/ODM/PHPCR/EventListener/DefaultParentListener.php [hunk 2]
```diff
@@ -28,14 +28,11 @@ class DefaultParentListener
     /** @var DocumentManagerInterface */
     private $documentManager;
```
Comment (Roshyo) [inline src/Bundle/Doctrine/ODM/PHPCR/EventListener/DefaultParentListener.php:29]: Why not enforcing this one too ?

**Pull request closed without merging**